### PR TITLE
Remove supports_dynamic_linker_feature from macOS unix crosstool

### PR DESCRIPTION
**Pull request by keith**
This feature isn't supported on macOS, and it isn't part of the Xcode CC toolchain, but it is left over here from when it was supported in the past.

https://github.com/bazelbuild/bazel/commit/ec5553352f2f661d39ac4cf665dd9b3c779e614c

Fixes: https://github.com/bazelbuild/bazel/pull/16414#issuecomment-1447374727

### CODE REVIEW

**Comment by keith** [comment]
I also have this fix in https://github.com/bazelbuild/bazel/pull/16619 but we can merge this first if we can't merge this one soon

**Review by brentleyjones — APPROVED** [review]

**Review by googlewalt — APPROVED** [review]

**Comment by keith** [comment]
done in https://github.com/bazelbuild/bazel/pull/16619

**Pull request closed without merging**
